api/v1/categories/PkpCategoryCategoryController.php

```diff
@@ -0,0 +1,409 @@
+<?php
+
+/**
+ * @file api/v1/categories/PkpCategoryCategoryController.php
+ *
+ * Copyright (c) 2014-2025 Simon Fraser University
+ * Copyright (c) 2003-2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PkpCategoryCategoryController
+ *
+ * @ingroup api_v1_category
+ *
+ * @brief Handle API requests for category operations.
+ */
+
+namespace PKP\API\v1\categories;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\file\PublicFileManager;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Route;
+use PKP\category\Category;
+use PKP\components\forms\context\CategoryForm;
+use PKP\context\Context;
+use PKP\core\Core;
+use PKP\core\PKPBaseController;
+use PKP\core\PKPRequest;
+use PKP\file\TemporaryFile;
+use PKP\file\TemporaryFileManager;
+use PKP\security\authorization\CanAccessSettingsPolicy;
+use PKP\security\authorization\ContextAccessPolicy;
+use PKP\security\Role;
+
+class PkpCategoryCategoryController extends PKPBaseController
```

Comment:
I'm personally against prefixing classes with PKP, it was probably used in the past as a kind of namespace when namespaces didn't exist.
But let me tag @asmecher to confirm the pattern.

Reply:
Agreed, the `PKP` class prefix is a deprecated pattern from before there was namespacing. We shouldn't use it for new code (and should gradually remove it as we disturb old code with breaking changes).

Reply:
Thanks, I've dropped the PKP from the class name